qiskit/opflow/primitive_ops/pauli_sum_op.py

```diff
@@ -358,6 +358,15 @@ def to_native(x):
                 to_native(np.real_if_close(self.primitive.coeffs[0])) * self.coeff,
             )
         coeffs = np.real_if_close(self.primitive.coeffs)
+        if not self.primitive.coeffs.dtype == object:
+            coeffs = np.real_if_close(self.primitive.coeffs)
+        else:
+            coeffs = []
+            for coeff in self.primitive.coeffs:
+                if not isinstance(coeff, (Parameter, ParameterExpression)):
+                    coeffs.append(np.real_if_close(coeff).item())
+                else:
+                    coeffs.append(coeff)
```

Comment:
There looks like some duplication here between the unchanged line above this, and the new `if` block.